### ISSUE TITLE
Undefined canFetchMore in first render with useInfiniteQuery using initialData

### ISSUE THREAD

**Issue by pawelphilipczyk** [tannerlinsley/react-query#250]
When I'm passing in initialData to the useInfiniteQuery it looks like the canFetchMore is undefined even though the getFetchMore would be true.

In fact it looks like the getFetchMore is actually not running at this first render.

My initial data is just an array with one initial group of data. This looks to me as a bug.

BTW I really like this library and the paging and infinite scroll api is way much better and readable than useSWRPages :)

**Comment by tannerlinsley** [comment]
I see. Is it much different than this example? https://codesandbox.io/s/agitated-bardeen-m06zt

**Comment by pawelphilipczyk** [comment]
It’s quite similar except I’m also using some initial data and it’s fetched and rendered on server side first.

**Comment by timwachsener** [comment]
Also having this problem with relay style cursor pased pagination. I guess `getFetchMore` needs to be run once when `initialData` is present.

**Comment by pawelphilipczyk** [comment]
Exactly what I suspect as well that probably `getFetchMore` is not run in first render.

**Comment by timwachsener** [comment]
@pawelphilipczyk To get this working in the meantime you can refetch the query on first render:
```
  useEffect(() => {
    result.refetch({ force: true });
  }, []);
```

**Comment by tannerlinsley** [comment]
Fixed with #266

**Comment by cherniavskii** [comment]
Fixed by #266
